File: Web.Admin/src/api/services/Center/tenant/models/QueryTenantPagedOutput.ts
import type { CommonStatusEnum } from "@/api/enums/CommonStatusEnum";
import type { EditionEnum } from "@/api/enums/EditionEnum";
import type { TenantTypeEnum } from "@/api/enums/TenantTypeEnum";

/**
 * 获取租户分页列表输出
 */
export interface QueryTenantPagedOutput {
	/**
	 * 租户Id
	 */
	tenantId?: string;
	/**
	 * 租户编号
	 */
	tenantNo?: string;
	/**
	 * 租户编码
	 */
	tenantCode?: string;
	/**
	 * 
	 */
	status?: CommonStatusEnum;
	/**
	 * 租户名称
	 */
	tenantName?: string;
	/**
	 * 租户简称
	 */
	shortName?: string;
	/**
	 * 租户英文名称
	 */
	spellName?: string;
	/**
	 * 
	 */
	edition?: EditionEnum;
	/**
	 * 租户管理员账号Id
	 */
	adminAccountId?: string;
	/**
	 * 租户管理员名称
	 */
	adminName?: string;
	/**
	 * 租户管理员手机
	 */
	adminMobile?: string;
	/**
	 * 租户管理员邮箱
	 */
	adminEmail?: string;
	/**
	 * 租户管理员电话
	 */
	adminPhone?: string;
	/**
	 * 租户机器人名称
	 */
	robotName?: string;
	/**
	 * 
	 */
	tenantType?: TenantTypeEnum;
	/**
	 * Logo URL
	 */
	logoUrl?: string;
	/**
	 * 允许删除数据
	 */
	allowDeleteData?: boolean;
	/**
	 * 
	 */
	departmentName?: string;
	/**
	 * 
	 */
	createdUserName?: string;
	/**
	 * 
	 */
	createdTime?: string;
	/**
	 * 
	 */
	updatedUserName?: string;
	/**
	 * 
	 */
	updatedTime?: string;
	/**
	 * 
	 */
	rowVersion?: string;
}

File: Api.Server/src/Center.Service/Tenant/Dto/QueryTenantPagedOutput.cs
// Copyright © 2018-Present 小方
// SPDX-License-Identifier: Apache-2.0
// 
// 本文件依据 Apache License 2.0 授权，完整条款见仓库根目录 LICENSE。
// 本软件按“原样”提供，相关免责声明及责任限制以许可证及适用法律为准。
// 版权来源、合法使用与二次开发责任说明见仓库根目录 README.md。

using Fast.Center.Domain;

namespace Fast.Center.Service.Tenant.Dto;

/// <summary>
/// 获取租户分页列表输出
/// </summary>
public class QueryTenantPagedOutput : PagedOutput
{
    /// <summary>
    /// 租户Id
    /// </summary>
    public long TenantId { get; set; }

    /// <summary>
    /// 租户编号
    /// </summary>
    [SugarSearchValue]
    public string TenantNo { get; set; }

    /// <summary>
    /// 租户编码
    /// </summary>
    [SugarSearchValue]
    public string TenantCode { get; set; }

    /// <summary>
    /// 状态
    /// </summary>
    public CommonStatusEnum Status { get; set; }

    /// <summary>
    /// 租户名称
    /// </summary>
    [SugarSearchValue]
    public string TenantName { get; set; }

    /// <summary>
    /// 租户简称
    /// </summary>
    public string ShortName { get; set; }

    /// <summary>
    /// 租户英文名称
    /// </summary>
    public string SpellName { get; set; }

    /// <summary>
    /// 版本
    /// </summary>
    public EditionEnum Edition { get; set; }

    /// <summary>
    /// 租户管理员账号Id
    /// </summary>
    public long AdminAccountId { get; set; }

    /// <summary>
    /// 租户管理员名称
    /// </summary>
    public string AdminName { get; set; }

    /// <summary>
    /// 租户管理员手机
    /// </summary>
    public string AdminMobile { get; set; }

    /// <summary>
    /// 租户管理员邮箱
    /// </summary>
    public string AdminEmail { get; set; }

    /// <summary>
    /// 租户管理员电话
    /// </summary>
    public string AdminPhone { get; set; }

    /// <summary>
    /// 租户机器人名称
    /// </summary>
    public string RobotName { get; set; }

    /// <summary>
    /// 租户类型
    /// </summary>
    public TenantTypeEnum TenantType { get; set; }

    /// <summary>
    /// Logo URL
    /// </summary>
    public string LogoUrl { get; set; }

    /// <summary>
    /// 允许删除数据
    /// </summary>
    public bool AllowDeleteData { get; set; }
}
